xclim/indices/run_length.py

```diff
@@ -41,18 +44,23 @@ def use_ufunc(
       N-dimensional input array.
     dim: str
       The dimension along which to find runs.
+    lastday: bool
+      If lastday is False, da values are N on the first day of a run, where N is the length of that run,
+      and are NaN on the other days of the runs.
+      If lastday is True, values are N on the last day of a run, where N is the length of that run,
+      and are NaN on the other days of the runs.
 
     Returns
     -------
     bool
       If ufunc_1dim is "auto", returns True if the array is on dask or too large.
       Otherwise, returns ufunc_1dim.
     """
-    if ufunc_1dim == "from_context":
-        ufunc_1dim = OPTIONS[RUN_LENGTH_UFUNC]
-
-    if ufunc_1dim == "auto":
-        return not uses_dask(da) and (da.size // da[dim].size) < npts_opt
+    if not lastday:
+        if ufunc_1dim == "from_context":
+            ufunc_1dim = OPTIONS[RUN_LENGTH_UFUNC]
+        if ufunc_1dim == "auto":
+            return not uses_dask(da) and (da.size // da[dim].size) < npts_opt
```

Comment:
```suggestion
    if ufunc_1dim == "from_context":
        ufunc_1dim = OPTIONS[RUN_LENGTH_UFUNC]
    if ufunc_1dim == "auto":
        ufunc_1dim =  not uses_dask(da) and (da.size // da[dim].size) < npts_opt
    # Now ufunc_1dim has to be a bool.
    # return True only if lastday is False.
    return (not lastday) and ufunc_1dim
```

Reply:
In the previous code, if lastday was True, there was a risk of returning a string. This didn't cause any bugs because non-empty strings evaluate to True. Thus, this change shouldn't change the behaviour, only make it cleaner.